internal/cmd/byok/command_create.go

```diff
@@ -202,3 +204,16 @@ func renderAzureEncryptionPolicy(key *byokv1.ByokV1Key) (string, error) {
 
 	return strings.Join(az, "\n"), nil
 }
+
+// Best effort to remove the key version from the Azure Key ID if it is present
+// For any errors, return the original key ID as is
+// All further validation of the key ID is done by the BYOK API
+func removeKeyVersionFromAzureKeyId(keyId string) string {
```

Comment:
You should only have to call `strings.Split()` once here, and you might want to use `strings.SplitN()`. Also, can you unit test this? It's hard to tell what line 218 is doing.

Reply:
Forgot to add the unit test to the PR, incoming.

Reply:
Decided in offline discussion to keep as is.